src/modules/flight_mode_manager/tasks/AutoLineSmoothVel/FlightTaskAutoLineSmoothVel.cpp

```diff
@@ -209,15 +221,18 @@ void FlightTaskAutoLineSmoothVel::_updateTrajConstraints()
 	_position_smoothing.setMaxJerk({max_jerk, max_jerk, max_jerk}); // TODO : Should be computed using heading
 
 	if (_is_emergency_braking_active) {
-		// When initializing with large downward velocity, allow 1g of vertical
+		// When initializing with large velocity, allow 1g of horizontal
 		// acceleration for fast braking
-		_position_smoothing.setMaxAccelerationZ(9.81f);
-		_position_smoothing.setMaxJerkZ(9.81f);
+		_position_smoothing.setMaxAcceleration({9.81f, 9.81f, 9.81f});
+		_position_smoothing.setMaxJerk({9.81f, 9.81f, 9.81f});
 
 		// If the current velocity is beyond the usual constraints, tell
 		// the controller to exceptionally increase its saturations to avoid
 		// cutting out the feedforward
+		_constraints.speed_xy = math::max(math::max(fabsf(_position_smoothing.getCurrentVelocityX()),
```

Comment:
this was removed in https://github.com/PX4/PX4-Autopilot/pull/18598

I discussed about this with @MaEtUgR and a solution would be to initialize the saturation of the controller to the current velocity when starting and monotonically decrease them with the speed until reaching the parameters. This way, we don't need to tell the controller to "exceptionally increase the saturation".

Reply:
Ok. I'm going to rebase and delete this line. 

I guess you want to catch when `_vehicle_control_mode.flag_multicopter_position_control_enabled` gets set and then also use an acceleration of 1g to decrease the limit until it's within bounds? Or how exactly did you want to monotonically decrease it?

Reply:
> then also use an acceleration of 1g to decrease the limit until it's within normal bounds

No, just decrease it with the incoming setpoint. In absolute values, this gives:
```
initialize saturation = max(vel_sp, desired_saturation)

then, at each iteration:
saturation = max(min(saturation, vel_sp), desired_saturation)
```

Reply:
Ok, got it. Should I include it in this PR?

Reply:
No, we can do that separately